types/jwplayer/index.d.ts

```diff
@@ -317,6 +317,329 @@ declare namespace jwplayer {
     type NoParamEvent = 'adBlock' | 'beforeComplete' | 'complete' | 'beforePlay' | 'displayClick' | 'playlistComplete'
             | 'seeked' | 'remove';
 
+    type RequiredConfiguration = {
+        readonly file: string;
+    } | {
+        readonly playlist: PlaylistConfiguration;
+    };
+
+    type Configuration = RequiredConfiguration & {
```

Comment:
Shouldn't this be an interface?

Reply:
The purpose of this arrangement is to require either the property `playlist` or `file` to be set, but not both.